cuBERT_topic_modelling/tests/test_subwordtokenizer.py

```diff
@@ -0,0 +1,73 @@
+from transformers import AutoTokenizer, AutoModel
+import torch
+import numpy as np
+from cuBERTopic import gpu_BERTopic
+from sklearn.datasets import fetch_20newsgroups
+import cudf
+import pytest
+
+device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
+
+# Sentences we want sentence embeddings for
+
+# TODO: sentence number 11927 is being encoded different in AutoTokenizer
+# and cuDF's SubwordTokenizer, look into that - this is causing high margin of error.
+
+sentences = fetch_20newsgroups(subset="all")["data"][:10000]
+
+
+def run_embedding_creation_transformers(sentences):
+    # Load AutoModel from huggingface model repository
+    tokenizer = AutoTokenizer.from_pretrained(
+        "sentence-transformers/all-MiniLM-L6-v2"
+    )
+
+    # Tokenize sentences
+    encoded_input = tokenizer(
+        sentences,
+        padding=True,
+        truncation=True,
+        max_length=128,
+        return_tensors="pt"
+    )
+
+    # Mean Pooling - Take attention mask into account for correct averaging
+    def mean_pooling(model_output, attention_mask):
+        token_embeddings = model_output[
+            0
+        ]  # First element of model_output contains all token embeddings
+        input_mask_expanded = (
+            attention_mask
+            .unsqueeze(-1)
+            .expand(token_embeddings.size())
+            .float()
+        )
+        sum_embeddings = torch.sum(token_embeddings * input_mask_expanded, 1)
+        sum_mask = torch.clamp(input_mask_expanded.sum(1), min=1e-9)
+        return sum_embeddings / sum_mask
```

Comment:
Import from the file , the function should not be redefined.

Reply:
This function is for running embedding creation with transformers `AutoTokenizer`, do you want me to import it from the `cuBERTopic.py` file where a similar implementation exists? The reason I recreated this here was that, if not recreating then I need to define an object of class `gpu_BERTopic` in that function, which didn't seem like something we should do.

Reply:
I think when you revamp it to a different file then it should be fine as requested in https://github.com/rapidsai/rapids-examples/pull/41#discussion_r757968545